src/kernel.hpp

```diff
@@ -127,37 +135,55 @@ struct cvk_kernel_argument_values {
     }
 
     bool init() {
-        if (m_kernel->has_pod_arguments()) {
+        if (m_kernel->has_pod_buffer_arguments()) {
             auto buffer = m_kernel->allocate_pod_buffer();
             if (buffer == nullptr) {
                 return false;
             }
 
             m_pod_buffer = std::move(buffer);
+        } else if (m_kernel->has_pod_arguments()) {
```

Comment:
We will need a unit test to exercise the case where POD arguments don't fit in the push constant space passed to clspv. It should pass with default options, expect a failure (probably during kernel creation so we don't fail if none of the problematic kernels are created) when push constants are used and always pass later on when both are used automatically. Further down the line, we will need to extend this to cover cases where multiple buffers are used (SSBO and UBO).

Reply:
A unit test makes sense, but currently clvk will never enable pod push constants. Manually adding the option will cause clvk to fail (because -pod-ubo is also specified). This should become testable after clspv starts automating the generation of pod args.

Reply:
You could pass `-pod-ubo=0` but I agree it might not be worth spending time on testing intermediate states if the entire move to push constants spans a relatively small period of time.

Reply:
That option can only appear once per compile, so clspv would generate an error unless I change how the option is specified.

Reply:
Right, nothing directly testable at this stage then.